### ISSUE TITLE
Error processing an error response 

### ISSUE THREAD

**Issue by castortech** [vercel/ai#2698]
### Description

This is using the openai provider for openrouter, which seems to get a good number of status 429 at times.

When the error occurs, it is not reported correctly with a status 429 and the quota message. Instead it goes into a complex error that is impossible to report back the real cause to the user.

### Code example

Very standard:
```
const openrouter = createOpenAI({ 
      apiKey: profile.openrouter_api_key,
      baseURL: "https://openrouter.ai/api/v1",
      compatibility: 'strict',
})

const result = await streamText({
	model: openrouter(chatSettings.model),
	messages: messages,
	temperature: chatSettings.temperature,
        maxTokens: chatSettings.maxTokens ?? 4096,
});

return result.toTextStreamResponse();
```

Error when quota exceeded:
```
Error: aborted
    at connResetException (node:internal/errors:787:14)
    at abortIncoming (node:_http_server:793:17)
    at socketOnClose (node:_http_server:787:3)
    at Socket.emit (node:events:530:35)
    at TCP.<anonymous> (node:net:337:12)
    at TCP.callbackTrampoline (node:internal/async_hooks:130:17) {
  code: 'ECONNRESET'
}
 ⨯ uncaughtException: Error: aborted
    at connResetException (node:internal/errors:787:14)
    at abortIncoming (node:_http_server:793:17)
    at socketOnClose (node:_http_server:787:3)
    at Socket.emit (node:events:530:35)
    at TCP.<anonymous> (node:net:337:12)
    at TCP.callbackTrampoline (node:internal/async_hooks:130:17) {
  code: 'ECONNRESET'
}
 ⨯ uncaughtException: Error: aborted
    at connResetException (node:internal/errors:787:14)
    at abortIncoming (node:_http_server:793:17)
    at socketOnClose (node:_http_server:787:3)
    at Socket.emit (node:events:530:35)
    at TCP.<anonymous> (node:net:337:12)
    at TCP.callbackTrampoline (node:internal/async_hooks:130:17) {
  code: 'ECONNRESET'
}
 ⨯ Error: failed to pipe response
    at pipeToNodeResponse (/home/ubuntu/dev/chatbot-ui/node_modules/next/dist/server/pipe-readable.js:126:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async DevServer.runEdgeFunction (/home/ubuntu/dev/chatbot-ui/node_modules/next/dist/server/next-server.js:1250:13)
    at async NextNodeServer.handleCatchallRenderRequest (/home/ubuntu/dev/chatbot-ui/node_modules/next/dist/server/next-server.js:248:37)
    at async DevServer.handleRequestImpl (/home/ubuntu/dev/chatbot-ui/node_modules/next/dist/server/base-server.js:816:17)
    at async /home/ubuntu/dev/chatbot-ui/node_modules/next/dist/server/dev/next-dev-server.js:339:20
    at async Span.traceAsyncFn (/home/ubuntu/dev/chatbot-ui/node_modules/next/dist/trace/trace.js:154:20)
    at async DevServer.handleRequest (/home/ubuntu/dev/chatbot-ui/node_modules/next/dist/server/dev/next-dev-server.js:336:24)
    at async invokeRender (/home/ubuntu/dev/chatbot-ui/node_modules/next/dist/server/lib/router-server.js:174:21)
    at async handleRequest (/home/ubuntu/dev/chatbot-ui/node_modules/next/dist/server/lib/router-server.js:353:24)
    at async requestHandlerImpl (/home/ubuntu/dev/chatbot-ui/node_modules/next/dist/server/lib/router-server.js:377:13)
    at async Server.requestListener (/home/ubuntu/dev/chatbot-ui/node_modules/next/dist/server/lib/start-server.js:141:13) {
  [cause]: TypeValidationError [AI_TypeValidationError]: Type validation failed: Value: {"error":{"message":"{\n  \"error\": {\n    \"code\": 429,\n    \"message\": \"Resource has been exhausted (e.g. check quota).\",\n    \"status\": \"RESOURCE_EXHAUSTED\"\n  }\n}\n","code":429}}.
  Error message: [
    {
      "code": "invalid_union",
      "unionErrors": [
        {
          "issues": [
            {
              "code": "invalid_type",
              "expected": "array",
              "received": "undefined",
              "path": [
                "choices"
              ],
              "message": "Required"
            }
          ],
          "name": "ZodError"
        },
        {
          "issues": [
            {
              "code": "invalid_type",
              "expected": "string",
              "received": "undefined",
              "path": [
                "error",
                "type"
              ],
              "message": "Required"
            },
            {
              "code": "invalid_type",
              "expected": "string",
              "received": "number",
              "path": [
                "error",
                "code"
              ],
              "message": "Expected string, received number"
            }
          ],
          "name": "ZodError"
        }
      ],
      "path": [],
      "message": "Invalid input"
    }
  ]
      at safeValidateTypes (webpack-internal:///(rsc)/./node_modules/@ai-sdk/provider-utils/dist/index.mjs:228:14)
      at safeParseJSON (webpack-internal:///(rsc)/./node_modules/@ai-sdk/provider-utils/dist/index.mjs:271:12)
      ... 5 lines matching cause stack trace ...
      at ensureIsPromise (node:internal/webstreams/util:185:19)
      at writableStreamDefaultControllerProcessWrite (node:internal/webstreams/writablestream:1109:5)
      at writableStreamDefaultControllerAdvanceQueueIfNeeded (node:internal/webstreams/writablestream:1224:5) {
    cause: ZodError: [
      {
        "code": "invalid_union",
        "unionErrors": [
          {
            "issues": [
              {
                "code": "invalid_type",
                "expected": "array",
                "received": "undefined",
                "path": [
                  "choices"
                ],
                "message": "Required"
              }
            ],
            "name": "ZodError"
          },
          {
            "issues": [
              {
                "code": "invalid_type",
                "expected": "string",
                "received": "undefined",
                "path": [
                  "error",
                  "type"
                ],
                "message": "Required"
              },
              {
                "code": "invalid_type",
                "expected": "string",
                "received": "number",
                "path": [
                  "error",
                  "code"
                ],
                "message": "Expected string, received number"
              }
            ],
            "name": "ZodError"
          }
        ],
        "path": [],
        "message": "Invalid input"
      }
    ]
        at get error [as error] (webpack-internal:///(rsc)/./node_modules/zod/lib/index.mjs:699:31)
        at Object.eval [as validate] (webpack-internal:///(rsc)/./node_modules/@ai-sdk/provider-utils/dist/index.mjs:198:101)
        at safeValidateTypes (webpack-internal:///(rsc)/./node_modules/@ai-sdk/provider-utils/dist/index.mjs:222:37)
        at safeParseJSON (webpack-internal:///(rsc)/./node_modules/@ai-sdk/provider-utils/dist/index.mjs:271:12)
        at Object.transform (webpack-internal:///(rsc)/./node_modules/@ai-sdk/provider-utils/dist/index.mjs:477:13)
        at ensureIsPromise (node:internal/webstreams/util:185:19)
        at transformStreamDefaultControllerPerformTransform (node:internal/webstreams/transformstream:505:18)
        at transformStreamDefaultSinkWriteAlgorithm (node:internal/webstreams/transformstream:555:10)
        at Object.write (node:internal/webstreams/transformstream:360:14)
        at ensureIsPromise (node:internal/webstreams/util:185:19) {
      issues: [Array],
      addIssue: [Function (anonymous)],
      addIssues: [Function (anonymous)],
      errors: [Array]
    },
    value: { error: [Object] },
    [Symbol(vercel.ai.error)]: true,
    [Symbol(vercel.ai.error.AI_TypeValidationError)]: true
  }
```

**Comment by lgrammel** [comment]
It seems OpenRouter is not fully OpenAI compatible. They should be returning a 429 instead of sending custom error messages in the stream.

**Comment by castortech** [comment]
The error is there:
{"error":{"message":"{\n  \"error\": {\n    \"code\": 429,\n    \"message\": \"Resource has been exhausted (e.g. check quota).\",\n    \"status\": \"RESOURCE_EXHAUSTED\"\n  }\n}\n","code":429}}

**Comment by castortech** [comment]
I would just add that Openrouter directly lists Vercel as a recommended client:

> We recommend the following clients:
>     [eventsource-parser](https://github.com/rexxars/eventsource-parser)
>     [OpenAI SDK](https://www.npmjs.com/package/openai)
>     [Vercel AI SDK](https://www.npmjs.com/package/ai)

**Comment by lgrammel** [comment]
Fix: https://github.com/vercel/ai/pull/2720

**Comment by lgrammel** [comment]
Fixed in `@ai-sdk/openai@0.0.49`

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This describes incorrect existing error handling in the OpenAI provider, where a meaningful streamed provider error was lost and replaced by an unrelated connection failure. The confirmed fix strongly supports classification as a bug.

### Evidence
- A streamed quota-exceeded response containing error code 429 was surfaced as ECONNRESET/“failed to pipe response” instead of exposing the actual quota error.
- The maintainer linked PR #2720 as a fix and confirmed it was released in @ai-sdk/openai@0.0.49.
